### ISSUE TITLE
Any plans for scipy.optimize?

### ISSUE THREAD

**Issue by Rolandw0w** [cupy/cupy#6112]
### Description

I wonder if there are any plans/ETAs on [scipy.optimize module](https://docs.scipy.org/doc/scipy/reference/optimize.html)?
Thank you!

**Comment by leofang** [comment]
There are some discussions ongoing but no concrete plan yet. @Rolandw0w Is there a particular functionality in the `scipy.optimize` module that you'd like to have?

cc: @cjnolet for vis

**Comment by Rolandw0w** [comment]
The problem that brought me here is linear programming and its slowness for scipy implementation.
I thought of RAPIDS and cuml; I've successfully used them on Kaggle earlier, so I started looking for GPU implementations.
Thanks!

**Comment by cjnolet** [comment]
@Rolandw0w, 

Indeed, the current plan is to generalize some of the solvers from cuml and move them to the RAFT project (https://github.com/rapidsai/raft)  so they can be exposed through the `cupy.optimize` package.

**Comment by Rolandw0w** [comment]
Sounds good, thank you!
Feel free to close this issue then.

**Comment by kmaehashi** [comment]
Let's keep this issue open as other can find the latest status 😃

**Comment by leofang** [comment]
xref: https://github.com/rapidsai/raft/issues/374

**Comment by cjnolet** [comment]
@Rolandw0w I just want to follow up on this issue as we are now starting to make RAFT features available to cupy through `pylibraft`. If you had your choice, is there any particular linear programming `method` you'd prefer to see implemented first?

**Comment by joeymuchacho** [comment]
> @Rolandw0w I just want to follow up on this issue as we are now starting to make RAFT features available to cupy through `pylibraft`. If you had your choice, is there any particular linear programming `method` you'd prefer to see implemented first?

@cjnolet Hi, I'm not Rolandw0w but I'd really like to have an implementation of SLSQP please (https://docs.scipy.org/doc/scipy/reference/optimize.minimize-slsqp.html).

**Comment by Rolandw0w** [comment]
@cjnolet If I had to choose, I'd prefer `HiGHS dual simplex`. IIRC, this method is suitable for parallel implementation.
But any other method would be great too.

**Comment by leofang** [comment]
In addition to SLSQP, CG, BFGS and L-BFGS-B are also great options for starter.

**Comment by cjnolet** [comment]
Tagging @teju85 for further viz.

**Comment by domattioli** [comment]
Any update?

**Comment by ndjvisser** [comment]
Any update on this? Would love to have a gpu version of scipy.optimize.minimize using SLSQP

**Comment by taylorpatti** [comment]
Chiming in to agree with ndjvisser - I'd love a scipy.optimize.minimize implementation.

**Comment by lpsgverrilla** [comment]
Don't know if possible, but I would like very much scipy.optimize.differential_evolution on cupy!

**Comment by jakirkham** [comment]
Maybe folks asking for this could share more about their use cases? How would you use this functionality? What would it help you accomplish?

**Comment by lpsgverrilla** [comment]
> Maybe folks asking for this could share more about their use cases? How would you use this functionality? What would it help you accomplish?

I don't have much knowledge about the subject at all, so I may be wrong in multiple accounts.
That said, recently I have started a deep quest into optimizing performance of a scipy.differential_evolution workflow (CPU).
One of the venues I explored was Differential Evolution in GPU, but I didn't find any python library offering it with same functionality of scipy's implementation, which is quite robust and rich.
I have found research papers citing how gpu could greatly improve performance of DE, though.
Currently I'm exploring other venues.

**Comment by craigwarner-ufastro** [comment]
Pasting my comment from #6324 here - I have developed a vectorized version of bounded value least squares based on scipy.optimize.bvls that I'd love to contribute.  Should I just go ahead and fork and submit a PR?  Just asking first because on #6324 it says ``` If you want to work on new modules (🥚), first discuss with us on this issue.```

https://github.com/cupy/cupy/issues/6324#issuecomment-2261191587

**Comment by michaelmcl1960** [comment]
I think a cupy version of least_squares would be extemely useful. https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.least_squares.html#scipy.optimize.least_squares

**Comment by richardrl** [comment]
+1 to @michaelmcl1960 . I am doing large-scale triangulation with nonlinear least squares. GPU would help a lot for this parallel computation.

**Comment by krishnapraveen7** [comment]
Bro what happened here? Is there any update?
